server/src/test/java/com/paypal/selion/grid/servlets/LoginServletTest.java

```diff
@@ -0,0 +1,262 @@
+     */
+    @Test
+    public void testPostForLogin() throws Exception {
+        MockHttpServletRequest request = new MockHttpServletRequest();
+        request.addParameter(LoginServlet.FORM_ID, "login");
+        request.addParameter(LoginServlet.USER_ID, "admin");
+        request.addParameter(LoginServlet.PASSWORD, "admin");
+        MockHttpServletResponse response = new MockHttpServletResponse();
+        servlet.doPost(request, response);
+        validateLoginSuccess(request, response, SeLionGridConstants.GRID_HOME_PAGE_URL);
+    }
+
+    /*
+     * HTTP POST based login action should fail
+     */
+    @Test
+    public void testPostForLoginFailure() throws Exception {
+        MockHttpServletRequest request = new MockHttpServletRequest();
+        request.getSession(true);
+        request.addParameter(LoginServlet.FORM_ID, "login");
+        request.addParameter(LoginServlet.USER_ID, "admin");
+        request.addParameter(LoginServlet.PASSWORD, "password");
+        MockHttpServletResponse response = new MockHttpServletResponse();
+        servlet.doPost(request, response);
+        validateLoginFailure(request, response);
+    }
+
+    /*
+     * HTTP POST based login action with valid credentials in session but invalid credentials in query parameters should
+     * succeed
+     */
+    @Test
+    public void testPostForLoginWithSessionCredentialsPresent() throws Exception {
+        MockHttpServletRequest request = new MockHttpServletRequest();
+        // add the valid credentials to the session
+        HttpSession session = request.getSession(true);
+        session.setAttribute(LoginServlet.USER_ID, "admin");
+        session.setAttribute(LoginServlet.PASSWORD, "admin");
+        // add invalid credentials to the query parameters
+        request.addParameter(LoginServlet.FORM_ID, "login");
+        request.addParameter(LoginServlet.USER_ID, "admin");
+        request.addParameter(LoginServlet.PASSWORD, "password");
+        MockHttpServletResponse response = new MockHttpServletResponse();
+        servlet.doPost(request, response);
+        validateLoginSuccess(request, response, SeLionGridConstants.GRID_HOME_PAGE_URL);
+    }
+
+    /*
+     * Test login action for selion sauce grid
+     */
+    @Test
+    public void testProcessForSauceGridLogin() throws Exception {
+        // Create a hubConfig with the Sauce capability matcher
+        GridHubConfiguration hubConfig = new GridHubConfiguration();
+        hubConfig.setCapabilityMatcher(new SeLionSauceCapabilityMatcher());
+
+        // Create a Selenium grid registry, using the new hubConfig
+        Registry registry = Registry.newInstance(null, hubConfig);
+
+        // Create a Selenium grid registration request
+        Map<String, Object> configuration = new HashMap<>();
+        configuration.put(RegistrationRequest.REMOTE_HOST, "http://127.0.0.1:2345");
+        configuration.put(RegistrationRequest.ID, "127.0.0.1");
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Do not hard code the IP address](https://www.codacy.com/app/SeLion/paypal-SeLion/file/2093723326/issues/source?bid=2928425&fileBranchId=3141724#l243)